overlord/devicestate/devicemgr.go

```diff
@@ -885,10 +885,30 @@ func (m *DeviceManager) RequestSystemAction(systemLabel string, action SystemAct
 		return ErrUnsupportedAction
 	}
 
-	// TODO:UC20 assume we are in the run mode
-	if sysAction.Mode == "run" {
-		// do nothing
-		return nil
+	switch m.systemMode {
```

Comment:
I think we could simplify the switch a bit:
```
	switch {	
	case m.systemMode == "":
		// probably test device manager mocking problem, or also potentially
		// missing modeenv
		return fmt.Errorf("internal error: unset system model")
	case m.systemMode == sysAction.Mode && systemLabel == currentSys.System: 
		// going from a mode and a system to the same mode and a system does nothing
		return nil
	case m.systemMode == "install":
		// it is not possible to request a system action from "install" mode,
		// we should let install mode finish first
		return ErrUnsupportedAction
	}
```
Anyways the current code looks correct too.